app/graphql/mutations/delete_mutation.rb

```diff
@@ -0,0 +1,26 @@
+module Mutations
+  class DeleteMutation < BaseMutation
+    argument :id, ID, required: true
+
+    field :id, ID, 'The deleted object ID.', null: false
+    field :errors, [Types::AttributeError], null: false
+
+    def resolve(id:)
+      object = load_object_by(id: id)
+      authorize!(object, :destroy)
+
+      User.as(context[:current_user]) do
+        errors = if object.destroy
+                   []
+                 else
+                   map_errors_to_path(resource)
```

Comment:
I think this should be `object` as `resource` is not defined. But when I change it, I get:

```
2019-04-01T08:15:45 [I|aud|a4efced6] Model (4) destroy event on name Blueberry
2019-04-01T08:15:45 [I|aud|a4efced6] Model (4) destroy event on info berry hw model updated
2019-04-01T08:15:45 [I|aud|a4efced6] Model (4) destroy event on vendor_class Awesome vendor
2019-04-01T08:15:45 [I|aud|a4efced6] Model (4) destroy event on hardware_model 4
2019-04-01T08:15:45 [E|app|a4efced6] You may not destroy Blueberry as it is in use!
2019-04-01T08:15:45 [I|app|a4efced6] Current user set to admin (admin)
2019-04-01T08:15:45 [I|app|a4efced6] Completed 500 Internal Server Error in 49ms (ActiveRecord: 8.7ms)
 | 
 | 
2019-04-01T08:15:45 [F|app|a4efced6]   
2019-04-01T08:15:45 [F|app|a4efced6] NoMethodError (undefined method `object' for #<Hash:0x007fbc448f25e8>
 | Did you mean?  object_id):
2019-04-01T08:15:45 [F|app|a4efced6]   
2019-04-01T08:15:45 [F|app|a4efced6] app/controllers/api/graphql_controller.rb:47:in `execute_single_graphql_query'
 | app/controllers/api/graphql_controller.rb:18:in `execute'
 | app/controllers/concerns/foreman/controller/timezone.rb:10:in `set_timezone'
 | app/models/concerns/foreman/thread_session.rb:32:in `clear_thread'
 | lib/foreman/middleware/telemetry.rb:10:in `call'
 | lib/foreman/middleware/catch_json_parse_errors.rb:9:in `call'
 | lib/foreman/middleware/logging_context_session.rb:22:in `call'
 | lib/foreman/middleware/logging_context_request.rb:11:in `call'
 | /home/vagrant/projects/katello/lib/katello/prevent_json_parsing.rb:12:in `call'

```